3party/opening_hours/opening_hours_tests/opening_hours_tests.cpp

```diff
@@ -112,11 +112,41 @@ bool IsOpen(osmoh::TRuleSequences const & rules, std::string const & dateTime)
   return GetRulesState(rules, dateTime) == osmoh::RuleState::Open;
 }
 
+std::string GetNextTimeOpen(osmoh::TRuleSequences const & rules, std::string const & dateTime)
+{
+  static auto const & fmt = "%Y-%m-%d %H:%M";
+  std::tm time = {};
+  if (!GetTimeTuple(dateTime, fmt, time))
+    throw GetTimeError{"Can't parse " + dateTime + " against " + fmt};
+
+  time_t openingTime = osmoh::GetNextTimeOpen(rules, mktime(&time));
+  tm openingTime_tm = *localtime(&openingTime);
+  char buffer[30];
+  std::strftime(buffer, 30, "%Y-%m-%d %H:%M", &openingTime_tm);
+
+  return std::string(buffer);
+}
+
 bool IsClosed(osmoh::TRuleSequences const & rules, std::string const & dateTime)
 {
   return GetRulesState(rules, dateTime) == osmoh::RuleState::Closed;
 }
 
+std::string GetNextTimeClosed(osmoh::TRuleSequences const & rules, std::string const & dateTime)
+{
+  static auto const & fmt = "%Y-%m-%d %H:%M";
```

Comment:
If the same fmt is used in both functions, please avoid copy-paste by defining it outside functions like kDateTimeFormat.

Reply:
Also it makes sense to remove copy-paste from other functions in this file.